web/client/reducers/mapInfo.js

```diff
@@ -80,10 +82,38 @@ function receiveResponse(state, action, type) {
                 responses[requestIndex] = updateResponse;
             }
         }
+        let firstLayerId = state.firstLayerId;
+        let indexObj;
+        const allResponsesLoaded = responses.filter(r => !isUndefined(r)).length === requests.length;
+        if (!isHover) {
+            // pull out the recievedResponses
+            const recievedResponses = responses.filter((res) => res);
+            if (recievedResponses.length && !firstLayerId) {
+                const format = state.requests[requestIndex]?.request?.info_format;
+                const layer = responses[requestIndex];
+                const validator = getValidator(format);
+                const valid = validator?.getValidResponses(recievedResponses);
+                if (valid.length) {
+                    firstLayerId = layer?.layerMetadata?.title || "";
+                }
+            }
+            if (allResponsesLoaded) {
```

Comment:
As per this implementation, it waits for all response to complete before showing the result in GFI panel. I believe the requirement was to show the first response as soon as it is received, because some servers might take time to respond. 
Your thoughts @tdipisa @MV88

Reply:
@dsuren1 yes, the first element for which we receive a response must be selected. This was the original requirement @dsuren1 you should know about that. It is also reported in the description of issue connected to this PR.
Is there a blocker with this? @dsuren1 @MuhweziDeo

Reply:
In  a way it is a blocker because currently unable to prevent `the menu from auto switches the selection`. as reported by @dsuren1 because I have to wait for all the responses. The first response received will be selected by the loader will not stop and the element won't be selected until all responses are received.